fog/view/server/src/fog_view_service.rs

```diff
@@ -151,6 +170,35 @@ impl<E: ViewEnclaveProxy, DB: RecoveryDb + Send + Sync> FogViewService<E, DB> {
         })
     }
 
+    fn process_queries(
+        &mut self,
+        fog_view_store_uri: FogViewStoreUri,
+        queries: Vec<attest::Message>,
+    ) -> MultiViewStoreQueryResponse {
+        let mut response = MultiViewStoreQueryResponse::new();
+        response.set_fog_view_store_uri(fog_view_store_uri.url().to_string());
+        for query in queries.into_iter() {
+            let result = self.query_impl(query);
+            if let Ok(attested_message) = result {
+                {
+                    let shared_state = self.db_poll_shared_state.lock().expect("mutex poisoned");
```

Comment:
*s there a reason to more narrowly scope the `db_polled_shared_state.lock()`?  it seems like unlocking at the brace or after `return response` shouldn't matter much.  It may be a convention I'm not aware of.  I see https://github.com/mobilecoinfoundation/mobilecoin/blob/master/CONTRIBUTING.md#avoid-manual-drops, but it says to use scopes over manual drops.  I don't think the intent is to always use scopes for lock guards

Reply:
I only want to unlock the state if the `attested_message` result is OK.